File: Spiral Matrix/ans.py
class Solution:
    def spiralOrder(self, matrix: List[List[int]]) -> List[int]:
        srow = 0
        erow = len(matrix) - 1
        scolumn = 0
        ecolumn = len(matrix[0]) - 1

        ans = []

        while srow <= erow and scolumn <= ecolumn:
            
            for i in range(scolumn, ecolumn + 1):
                ans.append(matrix[srow][i])
            srow += 1

            
            for i in range(srow, erow + 1):
                ans.append(matrix[i][ecolumn])
            ecolumn -= 1

            
            if srow <= erow:
                for i in range(ecolumn, scolumn - 1, -1):
                    ans.append(matrix[erow][i])
                erow -= 1

            
            if scolumn <= ecolumn:
                for i in range(erow, srow - 1, -1):
                    ans.append(matrix[i][scolumn])
                scolumn += 1
        return ans

File: Spiral Matrix/ans.cpp
class Solution {
public:
    vector<int> spiralOrder(vector<vector<int>>& matrix) {
        int srow = 0;
        int erow = matrix.size()-1;
        int scolumn = 0;
        int ecolumn = matrix[0].size()-1;
        vector<int> ans;

        while (srow <= erow && scolumn <= ecolumn){
            for (int i = scolumn ; i <= ecolumn;i++ ){
                ans.push_back(matrix[srow][i]);
            }
            srow ++;

            for (int i = srow ; i <= erow;i++ ){
                ans.push_back(matrix[i][ecolumn]);
            }
            ecolumn --;

            if (srow <= erow){
                for (int i = ecolumn ; i >= scolumn;i-- ){
                    ans.push_back(matrix[erow][i]);
                }
                erow --;
            }
            
            if (scolumn <= ecolumn){
                for (int i = erow ; i >= srow;i-- ){
                    ans.push_back(matrix[i][scolumn]);
                }
                scolumn ++;
            }
        }
        return ans;
        
    }
};
